src/libopensc/card-openpgp.c

```diff
@@ -787,6 +788,14 @@ pgp_enumerate_blob(sc_card_t *card, struct blob *blob)
 		}
 		tag |= cla;
 
+		/* Awful hack for composite DOs that have
+		 * a TLV with the DO's id encompassing the
+		 * entire blob. Example: Yubikey Neo */
+		if (tag == blob->id) {
```

Comment:
What is happening here is that the original code seems to expect something like this:

Reading DO 0x65 (Cardholder Related Data):
0x5B (Name)
0x07 (7 bytes)
'Example'
0x5F 0x2D (Language)
0x02 (2 bytes)
'en'
...

In other words, the original code is expecting a plain concatenation of the simple DOs. What the YubiKey and other javacardopenpgp-derived code does is something like the following:

Reading DO 0x65 (Cardholder Related Data):
0x65
0x14 (The entire 0x65 DO has 20 bytes)
0x5B (Name, this is the payload of the 0x65 object)
0x07 (7 bytes)
'Example'
0x5F 0x2D (Language)
0x02 (2 bytes)
'en'
Other fields up to 20 bytes

So the added hack here simply unwraps the outermost layer.